src/lib/pickles/plonk_checks/scalars.ml

```diff
@@ -270,18 +654,977 @@ module Tick : S = struct
        ; mds = _
        ; endo_coefficient
        ; srs_length_log2 = _
-       ; vanishes_on_last_4_rows = _
-       ; joint_combiner = _
-       ; beta = _
-       ; gamma = _
+       ; vanishes_on_last_4_rows
+       ; joint_combiner
+       ; beta
+       ; gamma
        ; unnormalized_lagrange_basis = _
-       ; enabled_if
+       ; if_feature
        ; foreign_field_modulus
        ; neg_foreign_field_modulus
        } :
         a Env.t ) =
     Column.Table.of_alist_exn
-      [ ( Index CompleteAdd
+      [ ( LookupKindIndex Xor
+        , lazy
+            (if_feature
+               ( LookupTables
+               , (fun () ->
+                   alpha_pow 24
+                   * ( vanishes_on_last_4_rows
+                     * ( field
+                           "0x40000000000000000000000000000000224698FC094CF91B992D30ED00000000"
+                       * ( cell (var (LookupAggreg, Curr))
+                         * ( ( field
+                                 "0x40000000000000000000000000000000224698FC094CF91B992D30ED00000000"
+                               * if_feature
+                                   ( LookupPattern Xor
```

Comment:
If we are doing this for each of the patterns, perhaps we want to have a function for each of them, so the chain is not so long, more modular